### PR TITLE
New TypeDoc option "markedOptions"

### LINKED ISSUE #1395: Disable email address escaping with HTML character references
## Search Terms

Markedjs, Marked, email, mangle

## Problem

Autolinked email addresses in the generated files are escaped with HTML character references.
On every run of TypeDoc, the escape sequence is randomly generated.
The result is, that the generated HTML files differ from the previous version, even if the meaningful content doesn't change. A version management system detects that as change then.

## Suggested Solution

There should be a way to disable atleast the `mangle` option of Marked (see https://marked.js.org/using_advanced).

Even better, if all Marked options would be configurable in the `typedoc.json` like:
```
"markedOptions": {
  "mangle": false
}
```

### PR DESCRIPTION
**Pull request by krisztianb**
Closes #1395

### CODE REVIEW

File: src/lib/output/plugins/MarkedPlugin.ts [hunk 1]
```diff
@@ -219,6 +215,34 @@ export class MarkedPlugin extends ContextAwareRendererComponent {
...
+     *
+     * @returns The options object for the markdown parser.
+     */
+    private createMarkedOptions(): Marked.MarkedOptions {
+        const markedOptions = (this.application.options.getValue(
+            "markedOptions"
+        ) ?? {}) as Marked.MarkedOptions;
+
+        if (
+            typeof markedOptions === "object" &&
```
Comment (Gerrit0) [inline src/lib/output/plugins/MarkedPlugin.ts:229]: We also need a check here for `null`

**Review by Gerrit0 — CHANGES_REQUESTED** [review]
I *really* dislike the theme tests... need to find a better solution. One small comment

File: src/lib/output/plugins/MarkedPlugin.ts (hunk 1 above)
Comment (krisztianb) [reply src/lib/output/plugins/MarkedPlugin.ts:229]: I think you missed the null coalescing a few lines above.

Comment (Gerrit0) [reply src/lib/output/plugins/MarkedPlugin.ts:229]: Oops! Looks like you're right, should be able to merge tonight.

**Comment by Gerrit0** [comment]
Thanks!

**Pull request merged**